src/util/models-universe.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  BoaviztaCpuImpactModel,
+  BoaviztaCloudImpactModel,
+  // CloudCarbonFootprint,
+  ShellModel,
+  SciMModel,
+  SciOModel,
+  TeadsCurveModel,
+} from '../lib';
+
+import {
+  GraphOptions,
+  ImplInitializeModel,
+  InitalizedModels,
+} from '../types/models-universe';
+
+/**
+ * Models Initialization Lifecycle.
+ */
+export class ModelsUniverse {
+  /**
+   * Models list.
+   */
+  public initalizedModels: InitalizedModels = {};
+
+  /**
+   * Gets model class by provided `name` param.
+   */
+  private handBuiltinModel(name: string) {
+    switch (name) {
+      case 'boavizta-cpu':
+        return BoaviztaCpuImpactModel;
+      case 'boavizta-cloud':
+        return BoaviztaCloudImpactModel;
+      // case 'ccf':
+      //   return CloudCarbonFootprint;
+      case 'teads-curve':
+        return TeadsCurveModel;
+      case 'sci-m':
+        return SciMModel;
+      case 'sci-o':
+        return SciOModel;
+      default: // cover default
+        return BoaviztaCpuImpactModel;
+    }
+  }
+
+  /**
+   * Returns plugin model.
+   */
+  private handPluginModel() {
+    return ShellModel;
+  }
+
+  /**
+   * Returns shell model.
+   */
+  private handShellModel() {
+    return ShellModel;
+  }
+
+  /**
+   * Gets model based on `kind` and `name` params.
+   */
+  private handModelByCriteria(kind: string, name: string) {
```

Comment:
If you want to handle plugin and shell you will also need to pass in path, since plugin and shell will need path to dynamically load.

Reply:
haven't handled that cases yet